src/lib/io/sol-iio.c

```diff
@@ -544,12 +544,27 @@ channel_get_pure_name(struct sol_iio_channel *channel)
 {
     size_t channel_name_len;
     char *channel_pure_name;
+    char *channel_name_suffix;
     bool modified = false;
 
     channel_name_len = strlen(channel->name);
-    if (channel_name_len > 2) {
-        char *channel_name_suffix;
+    if (channel_name_len > 5) {
+        channel_name_suffix = channel->name + channel_name_len - 5;
```

Comment:
No, it's not multi-channel logic. Look in this function, but some lines below. The job of stripping the Y component is already done. Just change the flow so it goes there after you stripped 'both' or 'ir' part.